ng-dev/misc/update-yarn/cli.ts

```diff
@@ -0,0 +1,125 @@
+import {spawnSync} from '../../utils/child-process';
+
+import {error, info, red} from '../../utils/console';
+import {Spinner} from '../../utils/spinner';
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client';
+import {addGithubTokenOption} from '../../utils/git/github-yargs';
+
+async function builder(yargs: Argv) {
+  return addGithubTokenOption(yargs);
+}
+
+/** Handles the command. */
+async function handler() {
+  /** Instance of the local git client. */
+  const git = AuthenticatedGitClient.get();
+  /** The main branch name of the repository. */
+  const mainBranchName = git.mainBranchName;
+
+  /** The original branch or ref before the command was invoked. */
+  const originalBranchOrRef = git.getCurrentBranchOrRevision();
+  if (git.hasUncommittedChanges()) {
+    error(
+      red(`Local working repository not clean. Please make sure there are no uncommitted changes.`),
+    );
+    process.exitCode = 1;
+    return;
+  }
+
+  const spinner = new Spinner('');
+  try {
+    spinner.update(`Fetching the latest primary branch from upstream: "${mainBranchName}"`);
+    git.run(['fetch', '-q', git.getRepoGitUrl(), mainBranchName]);
+    git.checkout('FETCH_HEAD', false);
+
+    /** Version numbers of the current (old) and new yarn installs. */
+    spinner.update(`Comparing current yarn version to the latest version on npm.`);
+    const yarnVersion = {
+      old: spawnSync('yarn', ['-v']).stdout.trim(),
+      new: spawnSync('yarn', ['info', 'yarn@latest', 'dist-tags.latest']).stdout.trim(),
+    };
+
+    if (yarnVersion.old === yarnVersion.new) {
+      spinner.complete();
+      error(red(`Yarn already updated to latest yarn (${yarnVersion.old})`));
+      process.exitCode = 1;
+      return;
+    }
+
+    /* File path of old yarn version */
+    const oldYarnPath = join(git.baseDir, `.yarn/releases/yarn-${yarnVersion.old}.cjs`);
+    const title = `build: update to yarn v${yarnVersion.new}`;
+    const body = `Update to the latest version of yarn, ${yarnVersion.new}.`;
+    const commitMessage = `${title}\n\n${body}`;
+    const {owner, repo} = git.remoteParams;
+    const branchName = `yarn-update-v${yarnVersion.new}`;
+    const localOwner = (await git.getForkOfAuthenticatedUser()).owner;
+
+    spinner.update(`Updating from yarn version '${yarnVersion.old}' to '${yarnVersion.new}'`);
+    spawnSync('yarn', ['policies', 'set-version', 'latest']);
+
+    spinner.update(`Removing previous yarn version (${yarnVersion.old})`);
+    unlinkSync(oldYarnPath);
+
+    spinner.update('Staging yarn vendoring files and creating commit');
+    git.run(['add', '.yarn/releases/**', '.yarnrc']);
+
+    /**
+     * New environment object with identifiers removed which prevent yarn from correctly loading
+     * new version from within a yarn call.
+     */
+    const env = {
+      ...process.env,
+      YARN_IGNORE_PATH: '0',
+      'PATH': process.env['PATH']!.replace(/\/tmp\/yarn.+?\:/, ''),
+    };
```

Comment:
Why is this custom environment needed? I think this needs more explanation in the code and I'm not sure if this is actually needed. can you please help clarifying that?